### PR TITLE
Add display of Sunsynk Timer setting to Battery

### PR DESCRIPTION
**Pull request by Papsnoek**
Howzit,
I added to the battery SOC to show the timer period capacity set and an charge icon to display if the battery will charge during this time period.
I wanted to add a Gen Charge and Grid Charge icon, but I battled with SVGs and space to add them, so settled on a single charge icon for both (Gen and/or Grid). You welcome to reorganize the display.

I also had fun trying to determine which timer setting is active according to the time now, so if you find a better/short way to do, again, feel free to change that code.

Thanks

image

### CODE REVIEW

**Comment by slipx06** [comment]
I've been thinking of doing this but the number of additional sensors and spacing put me off. I like what you did. Going to review the changes thanks.

File: dist/sunsynk-power-flow-card.js [hunk 1]
```diff
@@ -162,6 +163,150 @@ class SunsynkPowerFlowCard extends LitElement {
...
+        }
+      };
+      if (prog_time6 < timer_now) {
+        if (prog_time1 < timer_now) {
+          prog_time6.setTime(prog_time6.getTime() + (1*60*60*24*1000));
+        }
+      };
+
+      if (prog_time1 < timer_now && prog_time2 > timer_now) {
+        if ( prog1.charge.state === 'No Grid or Gen' ) {
```
Comment (slipx06) [inline dist/sunsynk-power-flow-card.js:266]: Are you using the kellarza integration ? We need to look at this sensor value is setup as a switch and sensor that returns a 0/1 if grid charge is enabled 

image

**Comment by slipx06** [comment]
Made a few adjustments. Thank for all the effort in making the card even better

**Comment by Papsnoek** [comment]
Thanks for Adding further checks :)
Yeah, I use kellarza integration, so all my checks were based on that. Also want to try using Solar Assistant but Kellarza is working well already. It is painful with the number of sensers.
Another enhancement I would like to make is to make the "Charge" icon clickable, so when your battery is below the required charge SOC, you can click the Charge Icon to update the Timer to start Charging.

**Comment by slipx06** [comment]
That should be easy enough to add. For now I just disable the system timer and it has the same effect charging to 100% though

**Comment by wacalitz** [comment]
Would something like that work with the E-Linter integration?  When I'm a bit behind on my SOC targets on bad weather days I just swap priority from load to battery to force grid usage and reach my SOC via available solar but need to do that in the Sunsynk app.  It would be great if the priority load/battery could also be a clickable toggle to trigger the modes but not sure this will work on the E-Linter integration without adding flows to node-red though.

**Comment by slipx06** [comment]
@wacalitz if you provide the card with a switch entity for these two attributes you can already toggle them from the card. For E-Linter you may need to modify the flows as I know they currently only read data and don't write data

 use_timer_248: switch.toggle_system_timer
 priority_load_243: switch.toggle_priority_load

**Pull request merged**